tools/test/scripts/doxy-spellchecker/spell.sh

```diff
@@ -0,0 +1,121 @@
+    # P4
+    # Keep me too
+    # Still here P1 But this shouldn't be P2
+    #_________________________________________
+    #
+    # Opted for having two separate formats. In particular this formatting issue came up when
+    # trying to strip the code segments and template type arguments between '<, >' as the multiline
+    # sed command would strip the entire line, causing the removal string to span across the entire file
+    # when trying to match the next end token (above format when stripping everything between P1 and P2
+    # would end up with just "Don't remove this" and the rest of the file stripped).
+
+    for ((i=0;i<${#start_tokens[@]};++i)); do
+        filter=""
+        if [[ "${formats[i]}" == 'strip_between' ]]; then
+            filter=$(echo "$res" | sed ""${start_tokens[i]}"/,"${end_tokens[i]}"/d")
+        elif [[ "${formats[i]}" == 'strip_between_sameline' ]]; then
+            filter=$(echo "$res" | sed -e "s/"${start_tokens[i]}".*"${end_tokens[i]}"//")
+        elif [[ "${formats[i]}" == 'strip_line' ]]; then
+            filter=$(echo "$res" | sed "/"${start_tokens[i]}"/ d")
+        elif [[ "${formats[i]}" == 'strip_token' ]]; then
+            filter=$(echo "$res" | sed "s/"${start_tokens[i]}"//g")
+        fi
+
+        if [ "$filter" != "" ]; then
+            res=$filter
+        fi
+    done
+
+    if [ "${2:-}" == "-vv" ]; then
+        echo "$res"
+    fi
+
+    prev_err=("")
+    while read err; do
+        if [ $(echo "$res" | grep "$err" | wc -l) -eq $(grep "$err" "$file" | wc -l) ]; then
+            # Do not count all caps words as errors (RTOS, WTI, etc) or plural versions (APNs/MTD's)
+            if ! [[ $err =~ ^[A-Z]+$ || $err =~ ^[A-Z]+s$ || $err =~ ^[A-Z]+\'s$ ]]; then
+
+                # Disregard camelcase/underscored words. Hex was stripped at the beginning
+                if ! echo "$err" | grep --quiet -E '[a-z]{1,}[A-Z]|_'; then
+
+                    # The grep command to fetch the line numbers will report all instances, do not
+                    # list repeated error words found from aspell in each file
+                    if ! [[ ${prev_err[*]} =~ "$err" ]]; then
+                        prev_err+=("$err")
+
+                        if [ ${#prev_err[@]} -eq 2 ]; then
+                            echo "================================="
+                            echo "Errors: "
+                        fi
+
+                        while read ln; do
+                            echo "$ln $err"
+                            ERRORS=$((ERRORS + 1))
+                        done <<< "$(grep -n "$err" "$file" | cut -d ' ' -f1)"
+                    fi
+                fi
+            fi
+        fi
+    done <<< "$(echo "$res" | aspell list -C --ignore-case -p "$DIR"/ignore.en.pws --local-data-dir "$DIR")"
+
+    if [ ${#prev_err[@]} -ne 1 ]; then
+        echo "_________________________________"
+    fi
+
+done <<< "$(find "$1" -type d -iname "*target*" -prune -o -name '*.h' -print)"
```

Comment:
You might be able to replace this with process substitution to make it quicker.

```suggestion
done < <(find "$1" -type d -iname "*target*" -prune -o -name '*.h' -print)
```